src/main/java/net/minecraftforge/event/entity/living/LivingFoodEffectEvent.java

```diff
@@ -0,0 +1,49 @@
+package net.minecraftforge.event.entity.living;
+
+import java.util.List;
+
+import com.mojang.datafixers.util.Pair;
+
+import net.minecraft.world.effect.MobEffectInstance;
+import net.minecraft.world.entity.LivingEntity;
+import net.minecraft.world.item.ItemStack;
+import net.minecraftforge.common.ForgeHooks;
+import net.minecraftforge.common.MinecraftForge;
+import net.minecraftforge.eventbus.api.Cancelable;
+
+/**
+ * LivingFoodEffectEvent is fire when an LivingEntity eats food item.<br>
```

Comment:
This is not descriptive on the point of this event and is not well named.

Reply:
I agree with the name point, but I could not think of any name that is descriptive and short. Do you have any suggestions?

Reply:
What about "LivingFoodEffectEvent is fire when an LivingEntity eats food item regardless of if any effects are applied."

Reply:
`LivingApplyEffectOnEatEvent` or probably `LivingApplyFoodEffectEvent`? It's just a thought that describes the action being taken.

As for your sentence, 'the event is fired whenever a LivingEntity receives an effect from food regardless of if the food applies any effects'. Though this brings a redundancy issue.